ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -35,19 +35,22 @@ class ProgressBar:
     def __init__(self):
         self.pbar = None
 
-    def _reset(self, engine, num_iterations):
+    def _reset(self, engine):
         self.pbar = tqdm(
-            total=num_iterations,
+            total=len(engine.state.dataloader),
             leave=False,
             bar_format='{desc}[{n_fmt}/{total_fmt}] {percentage:3.0f}%|{bar}{postfix} [{elapsed}<{remaining}]')
 
     def _close(self, engine):
         self.pbar.close()
 
-    def _update(self, engine, metric_names):
-        metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
+    def _update(self, engine, metric_names=None):
         self.pbar.set_description('Epoch {}'.format(engine.state.epoch))
-        self.pbar.set_postfix(**metrics)
+
+        if metric_names is not None:
+            metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
```

Comment:
Following the comment of @jasonkriss here: `engine.state.metrics[name]` can be the problem if `_update` is called on not `RunningAverage` metric.

Reply:
but if you set it in any other way, wouldn't you be able to access it? The metric must be set the progress bar, though

Reply:
you got me confused with the `_update` method call, since there is no such method. Also the `update` method in `RunningAverage` doesn't do anything

Reply:
Ah I got it... you mean the `_update` from the progress bar itself. 

That would not be called if there is no `ITERATION_COMPLETED` event fired...